runtime/lua/vim/secure.lua

```diff
@@ -86,21 +111,57 @@ function M.read(path)
     trust[fullpath] = hash
   end
 
+  write_trust(trust)
+
+  return contents
+end
+
+--- Update the trust status of file at path in the trust database at
+--- $XDG_STATE_HOME/nvim/trust.
+---
+--- @param path (string) Path to a file to update status for.
+--- @param allow (boolean) If true, set status to allow, otherwise set to deny.
+function M.trust(path, allow)
```

Comment:
We also want the ability to remove a file from the trust database completely, so this API will need to reflect that.

This could be represented as a `:trust forget <file>` command.

Reply:
Maybe separate functions - `M.allow`, `M.deny`, `M.forget` - and then make the ex command call a single "private" function like `M._command` that will parse the arguments?

Reply:
I like having a separate `trust` function as that makes it clear this is for managing the trust file. `vim.secure.allow` is a bit vague. Besides, `allow` and `deny` are just slight variations of the same concept, so grouping them together under a common interface makes sense to me.

Reply:
I think assuming `secure` would be responsible only for the limited access to certain files, like it is now, `vim.secure.allow` and `vim.secure.deny` are self-explanatory. I do prefer having a single `trust` function with a parameter as the public api, though.